test/test_tensor_ops.py

```diff
@@ -0,0 +1,411 @@
+                                     supports_out=False),
+                TensorCreationOpInfo('tensor_grad',
+                                     op=torch.tensor,
+                                     sample_inputs_func=sample_inputs_tensor,
+                                     supports_autograd=True,
+                                     supports_out=False),
+                TensorCreationOpInfo('tensor_grad_pinning',
+                                     op=torch.tensor,
+                                     sample_inputs_func=functools.partial(sample_inputs_tensor, pin_memory=True),
+                                     supports_autograd=True,
+                                     supports_out=False)]
+
+
+class TestTensorCreationOps(TestCase):
+
+    def _check_shares_memory(self, tensor, other):
+        """Check if two tensors share memory.
+        """
+        # Notice that tensors may share memory even when using
+        # different devices: recall pin_memory option
+        if tensor.layout == torch.strided:
+            if tensor.storage().data_ptr() == other.storage().data_ptr():
+                return True
+            # TODO: check for strides
+        return False
+
+    def _apply_samples(self, device, dtype, op, test_func):
+        samples = op.sample_inputs(device, dtype, requires_grad=op.supports_autograd)
+        exceptions = []
+        count = 0
+        for sample in samples:
+            count += 1
+            ref = sample.ref
+            args = sample.args if sample.input is sample.unspecified else (sample.input,) + sample.args
+            assert sample.ref is not sample.unspecified
+            if callable(ref):
+                ref = ref(*args, **sample.kwargs)
+            try:
+                if isinstance(ref, Exception):
+                    assert len(ref.args) == 1
+                    try:
+                        with self.assertRaisesRegex(type(ref), ref.args[0]):
+                            op(*args, **sample.kwargs)
+                    except Exception as exc:
+                        assert ref == exc
+                else:
+                    result = op(*args, **sample.kwargs)
+                    test_func(sample, result, ref)
+            except Exception as exc:
+                exceptions.append(f'{type(exc).__name__}:{exc} [{sample}]')
+                if count > 10 and len(exceptions) == count:
+                    # when all the first 10 samples have triggered
+                    # failures, stop the test as it is unlikely that
+                    # other samples do better
+                    exceptions.append('Test stopped after excessive failures: some samples may be untested')
+                    break
+        if exceptions:
+            msg = '\n  '.join(exceptions)
+            raise AssertionError(f'{len(exceptions)} samples out of {count} triggered failures\n  {msg}')
+
+    @suppress_warnings
+    @onlyOnCPUAndCUDA
+    @ops(tensor_op_db)
```

Comment:
fyi: If the ops were part of the op_db that iterable could still be filtered to just tensor creation ops.